packages/flutter/lib/src/cupertino/refresh.dart

```diff
@@ -0,0 +1,520 @@
+///
+/// See also:
+///
+///  * [CustomScrollView], a typical sliver holding scroll view this control
+///    should go into.
+///  * <https://developer.apple.com/ios/human-interface-guidelines/controls/refresh-content-controls/>
+///  * [RefreshIndicator], a Material Design version of the pull-to-refresh
+///    paradigm. This widget works differently than [RefreshIndicator] because
+///    instead of being an overlay on top of the scrollable, the
+///    [CupertinoRefreshControl] is part of the scrollable and actively occupies
+///    scrollable space.
+class CupertinoRefreshControl extends StatefulWidget {
+  /// Create a new [CupertinoRefreshControl] for inserting into a list of slivers.
+  ///
+  /// [refreshTriggerPullDistance], [refreshIndicatorExtent] both have reasonable
+  /// defaults and cannot be null.
+  ///
+  /// [builder] has a default indicator builder but can be null, in which case
+  /// no indicator UI will be shown but the [onRefresh] will still be invoked.
+  ///
+  /// [onRefresh] will be called when pulled far enough to trigger a refresh.
+  const CupertinoRefreshControl({
+    this.refreshTriggerPullDistance: _kDefaultRefreshTriggerPullDistance,
+    this.refreshIndicatorExtent: _kDefaultRefreshIndicatorExtent,
+    this.builder: buildSimpleRefreshIndicator,
+    this.onRefresh,
+  }) : assert(refreshTriggerPullDistance != null && refreshTriggerPullDistance > 0.0),
+       assert(refreshIndicatorExtent != null && refreshIndicatorExtent >= 0.0),
+       assert(
+         refreshTriggerPullDistance >= refreshIndicatorExtent,
+         'The refresh indicator cannot take more space in its final state '
+         'than the amount initially created by overscrolling'
+       );
+
+  /// The amount of overscroll the scrollable must be dragged to trigger a reload.
+  ///
+  /// Must not be null, must be larger than 0 and larger than [refreshIndicatorExtent].
+  ///
+  /// When overscrolled past this distance, [onRefresh] will be called if not
+  /// null and the [builder] will build in the [RefreshIndicatorMode.armed] state.
+  final double refreshTriggerPullDistance;
+
+  /// The amount of space the refresh indicator sliver will keep holding while
+  /// [onRefresh]'s [Future] is still running.
+  ///
+  /// Must not be null and must be positive. But can be 0, in which case the
+  /// sliver will start retracting back to 0 as soon as the refresh started.
+  ///
+  /// Must be smaller than [refreshTriggerPullDistance] since the sliver shouldn't
+  /// grow more after triggering the refresh.
+  final double refreshIndicatorExtent;
+
+  /// A builder that's called as this sliver's size changes and as the state
+  /// changes.
+  ///
+  /// A default simple Twitter-style pull-to-refresh indicator is provided if
+  /// not specified.
+  ///
+  /// Can be set to null, in which case nothing will be drawn in the overscrolled
+  /// space.
+  ///
+  /// Will not be called when the available space is 0 such as before any
```

Comment:
0 such --> zero, such

Reply:
Using digit 0 in the rest of the doc. Keeping for consistency

Reply:
Well, OK, but most writing style guides agree that numerals under 10 should be written out, rather than using digits, and since this is user-facing text, grammar and style is important.

Reply:
Seems like there are 98 instances of `0` in our dartdocs and 103 `zero` in our dartdocs :S

I changed the `0` in this file to `0.0` when I meant the double value exactly and `zero` when I meant the zero value conceptually like here.